### PR TITLE
Demo

### PR DESCRIPTION
**Pull request by cdaniluk**
Not really looking to make this perfect. In building this, I've realized that we absolutely _have_ to have some sort of layer in between the AWS events and the target to route appropriately. Everything is too coarse. There are so many influencers of whether an alert should wake you up, create a non-urgent alert, create a ticket, generate a slack notice or be discarded. But for now this shows TF's role in what we can do. I think we probably need to create a layer in between that talks to something like this, that speaks its language.

To be able to use this effectively, we'd need to make it support CloudWatch Alarms, Metric Filters (slightly different), CloudWatch Events and a couple of other things that aren't quite the same. AND figure out severity/routing on top of that.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,8 @@
 
 # .tfvars files
 *.tfvars
+
+*.zip
+build/*
+
+.DS_Store
```
Comment (sblack4) [inline .gitignore]: No newline at the EOF

File: jira.tf [hunk 2]
```diff
@@ -41,20 +68,48 @@ resource "aws_iam_role_policy_attachment" "jira" {
...
+}
+
+
+
 resource "aws_lambda_function" "jira" {
   count            = var.enable_jira_integration ? 1 : 0
   function_name    = local.jira_name
-  filename         = data.archive_file.jira.output_path
-  handler          = "create_jira.handler"
+  filename         = "${path.module}/create_jira.zip"
```
Comment (sblack4) [inline jira.tf:98]: We don't have a rule for this but if I reference an interpolated string more than once I like it to be a local

File: outputs.tf [hunk 3]
```diff
@@ -11,4 +11,4 @@ output "sns_topic_notify_arn" {
 output "sns_topic_ticket_arn" {
   description = "Ticketing Topic ARN"
   value       = aws_sns_topic.ticket.arn
-}
+}
```
Comment (sblack4) [inline outputs.tf]: no newline at the EOF

File: .gitignore (hunk 1 above)
Comment (sblack4) [inline .gitignore]:
```suggestion
.DS_Store

```

File: outputs.tf (hunk 3 above)
Comment (sblack4) [inline outputs.tf]:
```suggestion
}

```

**Review by sblack4 — CHANGES_REQUESTED** [review]
The build is failing because two files don't have newlines at the EOF. Other than that I looks 🔥 

pre-commit-hook output:
```
Fix End of Files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing outputs.tf
Fixing .gitignore
```

**Comment by dgoodellrhy** [comment]
> The build is failing because two files don't have newlines at the EOF. Other than that I looks 🔥
> 
> pre-commit-hook output:
> 
> ```
> Fix End of Files.........................................................Failed
> - hook id: end-of-file-fixer
> - exit code: 1
> - files were modified by this hook
> 
> Fixing outputs.tf
> Fixing .gitignore
> ```

Is there a good technical reason to block on missing newlines at EOF?  Does it break something?

**Comment by sblack4** [comment]
> > The build is failing because two files don't have newlines at the EOF. Other than that I looks 🔥
> > pre-commit-hook output:
> > ```
> > Fix End of Files.........................................................Failed
> > - hook id: end-of-file-fixer
> > - exit code: 1
> > - files were modified by this hook
> > 
> > Fixing outputs.tf
> > Fixing .gitignore
> > ```
> 
> Is there a good technical reason to block on missing newlines at EOF? Does it break something?

Is pedantry a good reason? It does break the build, but the build only checks the pre-commit hooks which don't do anything of substance. I'd argue that it's easy to just commit the suggested newlines through the PR UI so why not.

**Comment by cdaniluk** [comment]
> > > The build is failing because two files don't have newlines at the EOF. Other than that I looks 🔥
> > > pre-commit-hook output:
> > > ```
> > > Fix End of Files.........................................................Failed
> > > - hook id: end-of-file-fixer
> > > - exit code: 1
> > > - files were modified by this hook
> > > 
> > > Fixing outputs.tf
> > > Fixing .gitignore
> > > ```
> > 
> > 
> > Is there a good technical reason to block on missing newlines at EOF? Does it break something?
> 
> Is pedantry a good reason? It does break the build, but the build only checks the pre-commit hooks which don't do anything of substance. I'd argue that it's easy to just commit the suggested newlines through the PR UI so why not.

In my opinion, having a 0 lint policy means that we won't have module cleanups down the line, which keeps change sets to only what actually _changed_ in the module and not what was cleaned up from prior failed lint jobs.

So I'm okay hardlining on this particular type of pedantry.

**Review by sblack4 — APPROVED** [review]

**Comment by smiller171** [comment]
also @dgoodellrhy If you set up `pre-commit` on your local machine these errors will be automatically fixed locally before you ever push

**Comment by smiller171** [comment]
I've now updated the template repo so it'll automatically commit any changes from the linters

https://github.com/rhythmictech/terraform-anycloud-template/pull/8

File: README.md [hunk 4]
```diff
@@ -42,14 +42,17 @@ aws --region us-east-1 secretsmanager create-secret --name jira-api-token --secr
 <!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
 ## Requirements
 
-No requirements.
+| Name | Version |
+|------|---------|
+| terraform | >= 0.13.0 |
```
Comment (sblack4) [inline README.md:54]: 😾 I hope it's not hard to upgrade our everything

**Review by sblack4 — APPROVED** [review]
Is it ready for a release and the registry?

**Pull request merged**